engine/common/synchronization/engine.go

```diff
@@ -480,24 +497,23 @@ func (e *Engine) onSyncRequest(originID flow.Identifier, req *messages.SyncReque
 	}
 	err := e.con.Unicast(res, originID)
 	if err != nil {
-		return fmt.Errorf("could not send sync response: %w", err)
+		e.log.Warn().Err(err).Msg("sending sync response failed")
+		return nil
 	}
-
 	e.metrics.MessageSent(metrics.EngineSynchronization, metrics.MessageSyncResponse)
 
 	return nil
 }
 
 // onSyncResponse processes a synchronization response.
-func (e *Engine) onSyncResponse(originID flow.Identifier, res *messages.SyncResponse) {
-
+func (e *Engine) onSyncResponse(originID flow.Identifier, res *messages.SyncResponse) error {
```

Comment:
Why returning `error` if the error is always `nil`

Reply:
I branched off a version of your code prior to you changing this. I am happy to remove the error value. You are right, we don't need it here.

Reply:
same for https://github.com/onflow/flow-go/blob/1f0439f6f1a29a8841cab3bf539dc4c4e8167c99/engine/common/synchronization/engine.go#L609

will remove the error returns 👍

Reply:
cleaned up in https://github.com/onflow/flow-go/pull/1008/commits/cabc8ac7d0156cda146511b1733a0a3a020a54ed